test/bin/pyutils/build_bootc_images.py

```diff
@@ -0,0 +1,212 @@
+
+        if os.path.exists(cf_outdir):
+            common.print_msg(f"{cf_outdir} already exists")
+            if common.should_skip(cf_outname):
+                common.record_junit(groupdir, cf_path, "containerfile", "SKIPPED")
+                continue
+
+        common.print_msg(f"Processing {cf_path}")
+        common.run_command(
+            ["podman", "build", "-t", cf_outname, "-f", cf_path,
+                os.path.join(IMAGEDIR, "rpm-repos")], dry_run)
+        if os.path.exists(cf_outdir):
+            shutil.rmtree(cf_outdir)
+        common.run_command(["podman", "save", "--format", "oci-dir", "-o", cf_outdir, cf_outname], dry_run)
+
+
+def main():
+    # Parse command line arguments
+    parser = argparse.ArgumentParser(description="Process container files with Podman.")
+    parser.add_argument("-d", "--dry-run", action="store_true", help="Dry run: skip executing Podman commands.")
+    dirgroup = parser.add_mutually_exclusive_group(required=True)
+    dirgroup.add_argument("-l", "--layer-dir", type=str, help="Path to the layer directory to process.")
+    dirgroup.add_argument("-g", "--group-dir", type=str, help="Path to the group directory to process.")
+
+    args = parser.parse_args()
+    # Convert input directories to absolute paths
+    if args.group_dir:
+        args.group_dir = os.path.abspath(args.group_dir)
+    if args.layer_dir:
+        args.layer_dir = os.path.abspath(args.layer_dir)
+
+    try:
+        # Make sure the local RPM repository exists
+        if not os.path.isdir(LOCAL_REPO):
+            raise Exception("Run create_local_repo.sh before building images")
+
+        # Determine versions of RPM packages
+        set_rpm_version_info_vars()
+        # Prepare container lists for mirroring registries
+        common.delete_file(CONTAINER_LIST)
+        extract_container_images(SOURCE_VERSION, LOCAL_REPO, CONTAINER_LIST, args.dry_run)
+        # The following images are specific to layers that use fake rpms built from source
+        extract_container_images(f"4.{FAKE_NEXT_MINOR_VERSION}.*", NEXT_REPO, CONTAINER_LIST, args.dry_run)
+        extract_container_images(PREVIOUS_RELEASE_VERSION, PREVIOUS_RELEASE_REPO, CONTAINER_LIST, args.dry_run)
+        extract_container_images(YMINUS2_RELEASE_VERSION, YMINUS2_RELEASE_REPO, CONTAINER_LIST, args.dry_run)
+
+        # Process individual group directory
+        if args.group_dir:
+            process_containerfiles(args.group_dir, args.dry_run)
+            return
+        # Process a layer with group directories
+        for item in os.listdir(args.layer_dir):
+            item_path = os.path.join(args.layer_dir, item)
+            # Check if this item is a directory
+            if os.path.isdir(item_path):
+                process_containerfiles(item_path, args.dry_run)
+        common.print_msg("Build complete")
+    except Exception as e:
+        common.print_msg(f"An error occurred: {e}")
+        traceback.print_exc()
```

Comment:
I see no logs, is this sufficient information to find a problem in your opinion?

Reply:
I'm not sure if I like the idea of porting `set -x` - it produces massive logs that are cumbersome to read.